src/proto_unix.c

```diff
@@ -93,26 +93,28 @@ int get_unix_socket(void)
 
 void send_uevent(spnav_event *ev, struct client *c)
 {
-	int i, data[8] = {0};
+	int i, data[9] = {0};
 	float motion_mul;
 
 	if(lsock == -1) return;
 
 	switch(ev->type) {
 	case EVENT_MOTION:
 		data[0] = UEV_TYPE_MOTION;
+		data[1] = ev->dev;
```

Comment:
You can't do this. Shuffling the data fields, and increasing the length of the data, breaks the protocol.